script/foundry/utils/DeployHelper.sol

```diff
@@ -0,0 +1,369 @@
+import { AccessController } from "contracts/access/AccessController.sol";
+import { IPAccountImpl } from "contracts/IPAccountImpl.sol";
+import { IIPAccount } from "contracts/interfaces/IIPAccount.sol";
+import { IRoyaltyPolicyLAP } from "contracts/interfaces/modules/royalty/policies/IRoyaltyPolicyLAP.sol";
+import { Governance } from "contracts/governance/Governance.sol";
+import { AccessPermission } from "contracts/lib/AccessPermission.sol";
+import { Errors } from "contracts/lib/Errors.sol";
+import { PILFlavors } from "contracts/lib/PILFlavors.sol";
+// solhint-disable-next-line max-line-length
+import { DISPUTE_MODULE_KEY, ROYALTY_MODULE_KEY, LICENSING_MODULE_KEY, TOKEN_WITHDRAWAL_MODULE_KEY, CORE_METADATA_MODULE_KEY, CORE_METADATA_VIEW_MODULE_KEY } from "contracts/lib/modules/Module.sol";
+import { IPAccountRegistry } from "contracts/registries/IPAccountRegistry.sol";
+import { IPAssetRegistry } from "contracts/registries/IPAssetRegistry.sol";
+import { ModuleRegistry } from "contracts/registries/ModuleRegistry.sol";
+import { LicenseRegistry } from "contracts/registries/LicenseRegistry.sol";
+import { LicensingModule } from "contracts/modules/licensing/LicensingModule.sol";
+import { RoyaltyModule } from "contracts/modules/royalty/RoyaltyModule.sol";
+import { RoyaltyPolicyLAP } from "contracts/modules/royalty/policies/RoyaltyPolicyLAP.sol";
+import { DisputeModule } from "contracts/modules/dispute/DisputeModule.sol";
+import { ArbitrationPolicySP } from "contracts/modules/dispute/policies/ArbitrationPolicySP.sol";
+import { TokenWithdrawalModule } from "contracts/modules/external/TokenWithdrawalModule.sol";
+import { MODULE_TYPE_HOOK } from "contracts/lib/modules/Module.sol";
+import { IModule } from "contracts/interfaces/modules/base/IModule.sol";
+import { IHookModule } from "contracts/interfaces/modules/base/IHookModule.sol";
+import { IpRoyaltyVault } from "contracts/modules/royalty/policies/IpRoyaltyVault.sol";
+import { CoreMetadataModule } from "contracts/modules/metadata/CoreMetadataModule.sol";
+import { CoreMetadataViewModule } from "contracts/modules/metadata/CoreMetadataViewModule.sol";
+import { PILicenseTemplate, PILTerms } from "contracts/modules/licensing/PILicenseTemplate.sol";
+import { LicenseToken } from "contracts/LicenseToken.sol";
+
+// script
+import { StringUtil } from "./StringUtil.sol";
+import { BroadcastManager } from "./BroadcastManager.s.sol";
+import { StorageLayoutChecker } from "./upgrades/StorageLayoutCheck.s.sol";
+import { JsonDeploymentHandler } from "./JsonDeploymentHandler.s.sol";
+
+// test
+import { TestProxyHelper } from "test/foundry/utils/TestProxyHelper.sol";
+
+contract DeployHelper is Script, BroadcastManager, JsonDeploymentHandler, StorageLayoutChecker {
+    using StringUtil for uint256;
+    using stdJson for string;
+
+    ERC6551Registry internal immutable erc6551Registry;
+    IPAccountImpl internal ipAccountImpl;
+
+    // Registry
+    IPAccountRegistry internal ipAccountRegistry;
```

Comment:
Do not need this.

Reply:
`AccessControlled` requires `ipAccountRegistry` as parameter, and so far the codebase has used a separately deployed `ipAccountRegistry`. Should we replace this with `ipAssetRegistry` since it inherits?

Reply:
Resolved with 268d9c3 (pass in casted ipAssetRegistry)